### ISSUE TITLE
`paths.base` production build routing issue

### ISSUE THREAD

**Issue by inzanez** [sveltejs/kit#8091]
### Describe the problem

Hi all

I serve a Svelte application through a reverse proxy under a sub-path like `http://my-site.com/sub/path`. When forwarding the requests, the reverse proxy rewrites the path so that `/sub/path` gets removed, resulting in requets to the Svelte deployment as if it were hosted at the root.
To make this work, I set `paths.base` to `/sub/path`. Using the `node` adapter, it prepended that `paths.base` to every static asset URL so that it was correctly fetched.

After upgrading to the most recent version of `@sveltejs/kit` and `@adapter/node` things do not seem to work anymore. When setting the `paths.base` variable, it seems that the resulting `node` build expects me to hit the server with the routes `${paths.base}/actual-route`. Which is not really how it used to work,...

### Describe the proposed solution

Fix routing based on `paths.base`...

### Importance

nice to have

**Comment by dummdidumm** [comment]
- from what version did you upgrade?
- please provide a reproducible

**Comment by inzanez** [comment]
I did upgrade from:
```
"@sveltejs/adapter-node": "^1.0.0-next.46",
"@sveltejs/kit": "^1.0.0-next.166",
"svelte": "^3.46.4",
...
```
to

```
"@sveltejs/adapter-node": "^1.0.0-next.104",
"@sveltejs/kit": "^1.0.0-next.577",
"svelte": "^3.46.4",
```

I do not see how I could deliver a reproducible as it all depends on the proxy being in place and rewriting URLs,...?

**Comment by inzanez** [comment]
@dummdidumm Yes, it is quite a jump :-) Well, I will try and put something together, that might take some time though. I have just tried a fresh project and played around with the static adapter,...and I think I have seen some weird routing behavior as well (at least from what I am used to).

I will be back with a repo and more detailed information,...

**Comment by inzanez** [comment]
@dummdidumm 
https://github.com/inzanez/svelte-routing

There are two branches, `old-version` and `new-version`. You might register the difference between the two.
Have the final application using only relative paths so that it will fetch the right assets and work wherever it is hosted (path-wise,...so it might be 10 levels deep in a folder structure on an `nginx` server but it would still run).

**Comment by dummdidumm** [comment]
maybe you are seing the changed behavior due to this PR #4448 which fixed the longstanding issue #3726 . If so, then that's expected and you need to adjust your path-rewriting.

**Comment by inzanez** [comment]
@dummdidumm Yes, that seems to be the case :-) Absolutely agree with that.
